Add GQLstatus to the notifications of the HINT category (#150)
neo4j/docs-status-codes: modules/ROOT/pages/notifications/all-notifications.adoc

== `HINT` category

`HINT` notifications are returned by default when the Cypher planner or runtime cannot create a query plan to fulfill a specified hint, for example, `JOIN` or `INDEX`.

> This behavior of the Cypher planner or runtime can be changed by setting the configuration link:https://neo4j.com/docs/operations-manual/current/configuration/configuration-settings/#config_dbms.cypher.hints_error[`dbms.cypher.hints_error`] to `true` -- in which case the query will return an error instead.

Reviewer: ```suggestion
This behavior of the Cypher planner or runtime can be changed by setting the configuration link:https://neo4j.com/docs/operations-manual/current/configuration/configuration-settings/#config_dbms.cypher.hints_error[`dbms.cypher.hints_error`] to `true`.
In this case, the query will return an error.
```

m|HINT
|===

======
[.include-with-GQLSTATUS-code]
======

[cols="<1s,<4"]
|===
|GQLSTATUS code
m|01N30
|StatusDescription

> a|warn: join hint unfulfillable. Unable to create a plan with `JOIN ON $label`. Try to change the join key(s) or restructure your query.

Author: ```suggestion
a|warn: join hint unfulfillable. Unable to create a plan with `JOIN ON $var_list`. Try to change the join key(s) or restructure your query.
```

m|HINT
|===

======
[.include-with-GQLSTATUS-code]
======

[cols="<1s,<4"]
|===
|GQLSTATUS code
m|01N31
|StatusDescription

> a|warn: hinted index not found. Unable to create a plan with `$index` because the index does not exist.

Author: ```suggestion
a|warn: hinted index not found. Unable to create a plan with `$index_list` because the index does not exist.
```

The hinted index does not exist, make sure the label and property are spelled correctly.
If the spelling is correct, either create the index or remove the hint from the query.

======
[.include-with-GQLSTATUS-code]
======

Query::
+
[source, cypher]
----
MATCH (a: Label)
USING INDEX a:Label(id)
WHERE a.id = 1
RETURN a
----

Returned GQLSTATUS code::
01N31

Returned Status Description::
warn: hinted index not found.

> Unable to create a plan with `INDEX a:Label(id)` because the index does not exist.

Reviewer 2: ```suggestion
Unable to create a plan with `INDEX :Label(id)` because the index does not exist.
```

In the code it looks like the variable is not present here

The hinted index does not exist, please check the schema (index is: INDEX FOR ()-[`r`:`Rel`]-() ON (`r`.`id`))

Suggestions for improvement::
The hinted index does not exist, make sure the relationship type and property are spelled correctly.
If the spelling is correct, either create the index or remove the hint from the query.

======
[.include-with-GQLSTATUS-code]
======

Query::
+
[source, cypher]
----
MATCH ()-[r:Rel]-()
USING INDEX r:Rel(id)
WHERE r.id = 1
RETURN r
----

Returned GQLSTATUS code::
01N31

Returned Status Description::
warn: hinted index not found.

> Unable to create a plan with `INDEX r:Rel(id)` because the index does not exist.

Reviewer 2: ```suggestion
Unable to create a plan with `INDEX :Rel(id)` because the index does not exist.
```

In the code it looks like the variable is not present here
